src/NuGet.Clients/NuGet.CommandLine/MsBuildToolset.cs

```diff
@@ -111,15 +111,21 @@ public static string GetMsBuildDirFromVsDir(string vsDir)
                 return null;
             }
 
-            // Enumerate all versions of MSBuild present, take the highest
-            string msBuildDirectory = string.Empty;
-            var highestVersionRoot = Directory.EnumerateDirectories(msBuildRoot)
-                .OrderByDescending(ToFloatValue)
-                .FirstOrDefault(dir =>
-                {
-                    msBuildDirectory = SysPath.Combine(dir, "bin");
-                    return File.Exists(SysPath.Combine(msBuildDirectory, "msbuild.exe"));
-                });
+            // If "Current" MSBuild is available, it is the version to use
+            // see https://github.com/Microsoft/msbuild/issues/3778
+            string msBuildDirectory = SysPath.Combine(msBuildRoot, "Current", "bin");
```

Comment:
There are some usages in 
https://github.com/NuGet/NuGet.Client/blob/69a673582c62a63e917d152d871617ac5031dee1/src/NuGet.Core/NuGet.Protocol/Plugins/PluginDiscoveryUtility.cs#L34
as well, where given the MSBuild.exe we go up a few directories and then down into the nuget assembly directories to find the plugin. 

If the directory depth is the same, thne it should be fine however (it seems to be).

You can add a test with that path in [here](https://github.com/NuGet/NuGet.Client/blob/69a673582c62a63e917d152d871617ac5031dee1/test/NuGet.Core.Tests/NuGet.Protocol.Tests/Plugins/PluginDiscoveryUtilityTest.cs#L62)

Reply:
Yes, we added `Current` to keep the directory depth/relative paths the same.

Reply:
Added test with dev16-like paths.